File: article_61/c/test2.c
#include <SDL2/SDL.h>

#define WIDTH  640
#define HEIGHT 480

SDL_Window  *window;
SDL_Surface *primarySurface;
SDL_Surface *image;

int load() {
    SDL_Surface *tempImage;

    tempImage = SDL_LoadBMP("test1.bmp");

    if (!tempImage) {
        puts("Error loading image");
        return 0;
    }

    image = SDL_ConvertSurface(tempImage, primarySurface->format, 0);

    SDL_FreeSurface(tempImage);

    if (!image) {
        puts("Error converting surface");
        return 0;
    }
    return 1;
}

int init() {
    if (SDL_Init(SDL_INIT_VIDEO ) < 0) {
        puts("Error initializing SDL");
        return 0;
    } 

    window = SDL_CreateWindow("SDL2 example #2", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, WIDTH, HEIGHT, SDL_WINDOW_SHOWN);
    if (!window) {
        puts("Error creating window");
        return 0;
    }

    primarySurface = SDL_GetWindowSurface(window);
    if (!primarySurface) {
        puts("Error getting surface");
        return 0;
    }
    return 1;
}

void destroy() {
    SDL_FreeSurface(image);

    SDL_DestroyWindow(window);
    SDL_Quit();
}

int main(int argc, char** args) {
    if (!init()) {
        return 1;
    }
    if (!load()) {
        return 1;
    }

    SDL_BlitSurface(image, NULL, primarySurface, NULL);

    SDL_UpdateWindowSurface(window);
    SDL_Delay(5000);

    destroy();
    return 0;
}

File: article_61/go/test2.go
package main

import (
	"github.com/veandco/go-sdl2/img"
	"github.com/veandco/go-sdl2/sdl"
)

const (
	width  = 640
	height = 480
)

func main() {
	if err := sdl.Init(sdl.INIT_VIDEO); err != nil {
		panic(err)
	}
	defer sdl.Quit()

	window, err := sdl.CreateWindow("Example #2", sdl.WINDOWPOS_UNDEFINED, sdl.WINDOWPOS_UNDEFINED,
		width, height, sdl.WINDOW_SHOWN)
	if err != nil {
		panic(err)
	}
	defer window.Destroy()

	primarySurface, err := window.GetSurface()
	if err != nil {
		panic(err)
	}

	surfaceImage, err := img.Load("test1.bmp")
	if err != nil {
		panic(err)
	}
	defer surfaceImage.Free()

	primarySurface.FillRect(nil, sdl.MapRGB(primarySurface.Format, 192, 255, 192))

	err = surfaceImage.Blit(nil, primarySurface, nil)
	if err != nil {
		panic(err)
	}

	window.UpdateSurface()

	sdl.Delay(5000)
}